### PR TITLE
add sync ability in USARTs

### PR DESCRIPTION
**Pull request by andrewgazelka**
# Background

- needed sync USART capabilities

# Description

- add sync usart capabilities
- needed for `firmware/brain`

# Verification

- tested on logic analyzer.

### CODE REVIEW

File: src/serial.rs [hunk 1]
```diff
@@ -96,11 +97,32 @@ pub mod config {
...
         pub baudrate: Hertz,
         pub wordlength: WordLength,
         pub parity: Parity,
         pub stopbits: StopBits,
+        pub clock_phase: ClockPhase,
+        pub bit_order: BitOrder,
+        pub clock_polarity: ClockPolarity,
+
+        /// TODO: should we rename this?
+        pub clock_last_data_pulse: bool
```
Comment (stabler) [inline src/serial.rs]: The manual refers to this as `Last Bit CLock pulse` (LBCL)- can we rename to that and make it an enum like the others?

File: src/serial.rs [hunk 2]
```diff
@@ -96,11 +97,32 @@ pub mod config {
...
+        High,
+        Low
+    }
+
     pub struct Config {
         pub baudrate: Hertz,
         pub wordlength: WordLength,
         pub parity: Parity,
         pub stopbits: StopBits,
+        pub clock_phase: ClockPhase,
```
Comment (stabler) [inline src/serial.rs]: It looks like the naming convention in this struct is without the underscore

Comment (andrewgazelka) [reply src/serial.rs]: image
so something like this?

Comment (stabler) [reply src/serial.rs]: 👍

File: src/serial.rs [hunk 3]
```diff
@@ -420,7 +455,8 @@ macro_rules! usart {
                     usart: $USARTX,
                     config: impl Into<config::Config>,
                     prec: rec::$Rec,
-                    clocks: &CoreClocks
+                    clocks: &CoreClocks,
+                    sync: bool
```
Comment (sephamorr) [inline src/serial.rs:459]: rename uses of "sync" to "synchronous"

File: src/serial.rs [hunk 4]
```diff
@@ -96,11 +97,30 @@ pub mod config {
...
+        Msb
+    }
+
+    pub enum ClockPhase {
+        First,
+        Second
+    }
+
+    pub enum ClockPolarity {
+        High,
```
Comment (sephamorr) [inline src/serial.rs:111]: rename to Idle{High,Low} (I assume that's what they mean?)

File: src/serial.rs [hunk 5]
```diff
@@ -96,11 +97,30 @@ pub mod config {
         STOP1P5,
     }
 
+    pub enum BitOrder {
+        Lsb,
```
Comment (sephamorr) [inline src/serial.rs:101]: rename to {LsbMsb}First

**Review by sephamorr — CHANGES_REQUESTED** [review]
3 comments, otherwise looks good.

**Comment by andrewgazelka** [comment]
@sephamorr what do you want me to do since I already merged? feature/usart still has updates you requested. https://github.com/parallelsystems/stm32h7xx-hal/tree/feature/sync-usart

**Comment by sephamorr** [comment]
oops, sorry on the delay. can you make a new PR really quick and I'll approve it?

**Pull request merged**